invenio_files_rest/serializer.py

```diff
@@ -18,6 +19,44 @@
 from .models import Bucket, MultipartObject, ObjectVersion, Part
 
 
+class MarshalResultBase(object):
+    """Wrapper class to provide backward compatibility with marshmallow 2."""
+
+    def __new__(cls, result, errors=None, *args, **kwargs):
+        """Instanciate Marshal Result class."""
+        def data(self):
+            warnings.warn(
+                "Schema().dump().data attribute is deprecated "
+                "in marshmallow v3.x",
+                category=DeprecationWarning, stacklevel=2)
+            return result
+
+        def data_errors(self):
+            warnings.warn(
+                "Schema().dump().errors attribute is deprecated "
+                "in marshmallow v3.x",
+                category=DeprecationWarning, stacklevel=2)
+            return result
+
+        setattr(cls, 'data', property(data))
+        setattr(cls, 'errors', property(data_errors))
```

Comment:
after a thought, I would remove this and let it fail on errors, or I would raise an exception, but I will wait for a feedback